projects/packages/identity-crisis/src/class-ui.php

```diff
@@ -0,0 +1,150 @@
+	 */
+	public static function enqueue_scripts() {
+		if ( is_admin() ) {
+			Assets::register_script(
+				'jp_identity_crisis_banner',
+				'../build/index.js',
+				__FILE__,
+				array(
+					'in_footer'  => true,
+					'textdomain' => 'jetpack',
+				)
+			);
+			Assets::enqueue_script( 'jp_identity_crisis_banner' );
+			wp_add_inline_script( 'jp_identity_crisis_banner', static::get_initial_state(), 'before' );
+
+			add_action( 'admin_notices', array( static::class, 'render_container' ) );
+		}
+	}
+
+	/**
+	 * Create the container element for the IDC banner.
+	 */
+	public static function render_container() {
+		?>
+		<div id="jp-identity-crisis-container"></div>
+		<?php
+	}
+
+	/**
+	 * Return the rendered initial state JavaScript code.
+	 *
+	 * @return string
+	 */
+	private static function get_initial_state() {
+		return 'var JP_IDENTITY_CRISIS__INITIAL_STATE=JSON.parse(decodeURIComponent("' . rawurlencode( wp_json_encode( static::get_initial_state_data() ) ) . '"));';
+	}
+
+	/**
+	 * Get the initial state data.
+	 *
+	 * @return array
+	 */
+	private static function get_initial_state_data() {
+		$idc_urls       = Identity_Crisis::get_mismatched_urls();
+		$current_screen = get_current_screen();
+		$is_admin       = current_user_can( 'jetpack_disconnect' );
+
+		return array(
+			'WP_API_root'         => esc_url_raw( rest_url() ),
+			'WP_API_nonce'        => wp_create_nonce( 'wp_rest' ),
+			'wpcomHomeUrl'        => $idc_urls['wpcom_url'],
+			'currentUrl'          => $idc_urls['current_url'],
+			'redirectUri'         => str_replace( '/wp-admin/', '/', $_SERVER['REQUEST_URI'] ),
+			'tracksUserData'      => Jetpack_Tracks_Client::get_connected_user_tracks_identity(),
+			'tracksEventData'     => array(
+				'isAdmin'       => $is_admin,
+				'currentScreen' => $current_screen ? $current_screen->id : false,
+			),
+			'isSafeModeConfirmed' => Identity_Crisis::$is_safe_mode_confirmed,
+			'consumerData'        => static::get_consumer_data(),
+			'isAdmin'             => $is_admin,
+		);
+	}
+
+	/**
+	 * Get the package consumer data.
+	 *
+	 * @return array
+	 */
+	private static function get_consumer_data() {
+		$consumers = apply_filters( 'jetpack_idc_consumers', array() );
+
+		if ( ! $consumers ) {
+			return array();
+		}
+
+		usort(
+			$consumers,
+			function ( $c1, $c2 ) {
+				$priority1 = ( array_key_exists( 'priority', $c1 ) && (int) $c1['priority'] ) ? (int) $c1['priority'] : 10;
+				$priority2 = ( array_key_exists( 'priority', $c2 ) && (int) $c2['priority'] ) ? (int) $c2['priority'] : 10;
+
+				return $priority1 > $priority2 ? 1 : -1;
+			}
+		);
+
+		$consumer_chosen     = null;
+		$consumer_url_length = 0;
+
+		foreach ( $consumers as $consumer ) {
+			if ( ! array_key_exists( 'admin_page', $consumer ) || ! is_string( $consumer['admin_page'] ) ) {
+				continue;
+			}
+
+			if ( 0 === strpos( $_SERVER['REQUEST_URI'], $consumer['admin_page'] ) && strlen( $consumer['admin_page'] ) > $consumer_url_length ) {
```

Comment:
Thanks Kim!

I fixed the "empty string" notice, added the check as you suggested: https://github.com/Automattic/jetpack/pull/22060/commits/58ca4665aaa9be08be284153ac23f6494335306b

> The second condition is just checking that the $consumer['admin_page'] value isn't an empty string

It was supposed to be looking for the longest `admin_page`, but updating of the `consumer_url_length` value got lost somehow (or I forgot to add it in the first place 🤷), and we didn't notice because of the order in which plugins initialize the Identity Crisis package.
Thanks for noticing this, I added the missing line.

This condition is hard to test, but I'd appreciate if you could give it a look:
1. Add the Jetpack Backup IDC customization from the testing instructions.
2. Go to line 120 and replace the "greater than" with "less than": `return $priority1 > $priority2 ? 1 : -1;` becomes `return $priority1 < $priority2 ? 1 : -1;`.
3. Comment out the line 144: `$consumer_url_length = strlen( $consumer['admin_page'] );`.
4. Go to Jetpack Backup, you should not see the "Backup Safe Mode". The regular "Safe Mode" text should appear.
5. Uncomment the line 144, reload the Backup Dashboard. Now the "Backup Safe Mode" line should appear.

Reply:
I understand why we need the length now. I tested with these instructions, and it worked as expected. 😄